src/alt/store/index.js

```diff
@@ -53,7 +53,13 @@ export function createStoreConfig(globalConfig, StoreModel) {
         return state
       }
     },
-    setState: fn.assign
+    setState(currentState, nextState) {
+      if (Object.prototype.toString.call(nextState) === '[object Object]') {
```

Comment:
Yeah, but #344 opens a whole can of worms in how state is managed and exposes a lot of internals. I'm still conflicted on it.